lib/r10k/source/base.rb

```diff
@@ -65,4 +65,17 @@ def accept(visitor)
       end
     end
   end
+
+  private
+
+  def derive_prefix(name,prefix)
```

Comment:
So now that you've changed this to how I asked, I want you to change it again! Yay! :)

Per your comment on https://github.com/puppetlabs/r10k/pull/295/files#r24181790, the BranchName class will never be passed true, only false and a string. However the BranchName class is currently encapsulating most of the logic of naming branches, but this code touches branch naming as well. Instead of deriving the prefix here, why don't we push that into the BranchName class and just pass it through here? It'll help decouple things and prevent the Source classes from having to know too much about how prefixing is implemented.

In addition, both the Git and SVN sources have a BranchName method but I think there's no difference between them; it might be reasonable to extract the common code into `R10K::Source::BranchName` and deduplicate those classes.